js/modules/link_text.d.ts

```diff
@@ -0,0 +1,9 @@
+declare namespace LinkText {
+  type Attributes = {
+    [key: string]: string;
+  }
+}
+
+declare function linkText(value: string, attributes: LinkText.Attributes): string;
+
+export = linkText;
```

Comment:
Nice way to add inline type definitions for existing JS files, see: https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html#template